pkg/oci/load_balancer_security_lists.go

```diff
@@ -362,6 +378,23 @@ func getNodeIngressRules(
 	ingressRules := []core.IngressSecurityRule{}
 
 	for _, rule := range rules {
+		// Remove (do not re-add) any rule that represents the old case when
+		// mutating a single ranged backend port or health check port.
+		if rule.TcpOptions != nil && rule.TcpOptions.DestinationPortRange != nil &&
+			*rule.TcpOptions.DestinationPortRange.Min == *rule.TcpOptions.DestinationPortRange.Max &&
+			*rule.TcpOptions.DestinationPortRange.Min != desiredPorts.BackendPort && *rule.TcpOptions.DestinationPortRange.Max != desiredPorts.BackendPort &&
```

Comment:
This large logical conjunction looks quite prone to mistakes and I think is duplicated again below. Might be worth factoring this into a smaller helper function and unit testing.

Reply:
Yeagh, I mentioned this in the JIRA and I started something like this on the ('other branch')[https://github.com/oracle/oci-cloud-controller-manager/blob/f2a9f78c50cb231128c27ee23392db84bca0749e/pkg/oci/load_balancer_security_lists.go#L516]. However, I thought we were meant to be making this fix with a few changes possible and then iterating to improve other bits? I am happy to have a go at improving this now, but, was trying to avoid a 'large commit'. Do you want me to refactor this code in this branch along with the fix attempt?